src/api/java/com/minecolonies/api/colony/buildings/views/IBuildingView.java

```diff
@@ -209,4 +211,38 @@
      * @return true if so.
      */
     boolean isDeconstructed();
+
+    /**
+     * Get the first module view matching the class.
+     * @param clazz the class to match.
+     * @param <T> the type of module returned.
+     * @return the module or an empty optional.
+     */
+    @NotNull
+    <T extends IBuildingModuleView> Optional<T> getModuleView(Class<T> clazz);
+
+    /**
+     * Get the module view matching a certain predicate.
+     * @param clazz the class of the module.
+     * @param modulePredicate the predicate to match.
+     * @param <T> the optional type.
+     * @return an optional of the matching module (could be empty).
+     */
+    @NotNull
+    <T extends IBuildingModuleView> Optional<T> getModuleViewMatching(Class<T> clazz, Predicate<IBuildingModuleView> modulePredicate);
```

Comment:
You can make this an `Predicate<? extends T>` for the selector, which would reduce the amount of casts down the line.

Reply:
Ahhh, it's ? extends T, I tried T and it didn't work =D